### PR TITLE
Chen And Garry's Ice Cream: Ice Cream DLC (LIZARD APPROVED!)

### PR DESCRIPTION
**Pull request by YakumoChen**
## About The Pull Request

Authored with help and love from @Thalpy 

I scream for ice cream!!

image

Introduces many new flavours of ice cream:
-Caramel
-Banana
-Lemon Sorbet
-Orange Creamsicle
-Peach (Limited Edition!)
-Cherry chip
-Korta Vanilla (made with lizard-friendly ingredients!)

image

Korta Cones! Now too can Nanotrasen's sanitation staff enjoy the wonders of ice cream!
You can also substitute custom ice cream flavours with korta milk! Finally, the meaty ice cream lactose-intolerants asked for is in reach!

## Why It's Good For The Game

I always thought the ice cream vat could use more flavours. The custom flavour besides, it isn't as intuitive to rename the cone and the added variety is good. The lack of a banana flavour already was questionable.
All the ice cream flavours used a selection of five sprites, now it's just one sprite and better supporting more additions.
Some of the flavours don't use milk! You can't do this with the custom flavour, making it slightly more interesting.

## Changelog
:cl: YakumoChen, Thalpy
add: Chen And Garry's Ice Cream is proud to debut a wide selection of cool new frozen treat flavours on a space station near you!
add: Chen And Garry's Ice Cream revolutionary Korta Cones allow our ice cream vendors to profit off the lizard demographic like never before!
code: Ice cream flavours now are all greyscaled similarly to GAGs
/:cl:

### CODE REVIEW

**Comment by Y0SH1M4S73R** [comment]
Could use the ability to make custom lizard ice creams (think the custom recipe but with korta milk instead of regular milk)

**Comment by YakumoChen** [comment]
> Could use the ability to make custom lizard ice creams (think the custom recipe but with korta milk instead of regular milk) Better yet a toggle to use milk or korta milk to make normal or lizard-friendly ice cream.

the cone is the only thing that actually matters code-wise for lizard edibility, but sure, I'll add it

File: code/datums/components/food/ice_cream_holder.dm [hunk 1]
```diff
@@ -258,40 +260,85 @@ GLOBAL_LIST_INIT_TYPED(ice_cream_flavours, /datum/ice_cream_flavour, init_ice_cr
...
+	color = COLOR_ICECREAM_LEMON
+	desc = "filled with lemon sorbet. Tangy!"
+	ingredients = list(/datum/reagent/consumable/ice, /datum/reagent/consumable/lemonjuice) //contains no milk
+	reagent_type = /datum/reagent/consumable/lemonjuice
+
+/datum/ice_cream_flavour/caramel
+	name = ICE_CREAM_CARAMEL
+	color = COLOR_ICECREAM_CARAMEL
+	desc = "filled with caramel ice cream. It is deliciously sweet."
+	ingredients = list(/datum/reagent/consumable/milk, /datum/reagent/consumable/ice, /datum/reagent/consumable/coco)
```
Comment (Y0SH1M4S73R) [inline code/datums/components/food/ice_cream_holder.dm:293]: coco should be caramel here

Comment (YakumoChen) [reply code/datums/components/food/ice_cream_holder.dm:293]: fixed, thanks

**Comment by DATA-xPUNGED** [comment]
hello am lizard i approve

**Comment by Loafin34** [comment]
blue cherry ice cream please!!!!

**Comment by YakumoChen** [comment]
> blue cherry ice cream please!!!!

Would a chocolate cherry do? Blue cherries are more hassle than what I think should be most of the default flavours, and you can just make a custom flavour with just cherry jelly

**Comment by YesterdaysPromise** [comment]
Blue Mystery Ice Cream:  bluespace dust + blueberry. 50/50 it contains either.
Also perhaps herbs+chocolate for mint ice cream? That is my favourite flavour.
Also also maybe cotton-carmel ice cream for moths?
And jupiter caps ice cream for ethereals?

**Comment by YakumoChen** [comment]
> Blue Mystery Ice Cream: bluespace dust + blueberry. 50/50 it contains either. Also perhaps herbs+chocolate for mint ice cream? That is my favourite flavour. Also also maybe cotton-carmel ice cream for moths? And jupiter caps ice cream for ethereals?

I would need special cones for each race, and I don't think a moth cone would be as easy as it sounds

I don't know what 'herbs' is but it needs to be a reagent

**Comment by carshalash** [comment]
Matcha ice cream sisters... we lost again.

**Pull request merged**